### PR TITLE
Feature/selfupdate updates venv

### PR DESCRIPTION
**Pull request by evansde77**
Adds --upgrade to build command to allow build to update any dependencies in the requirements file via pip install --upgrade. 
Add the upgrade flag in the git cirrus build called by selfupdate

Addresses:
https://github.com/evansde77/cirrus/issues/98

### CODE REVIEW

**Comment by evansde77** [comment]
@ksnavely @dmannarino review?

File: src/cirrus/build.py [hunk 1]
```diff
@@ -45,6 +45,13 @@ def build_parser(argslist):
         nargs='*',
         help='generate documentation with Sphinx (Makefile path must be set in cirrus.conf.')
 
+    parser.add_argument(
+        '-u',
+        '--upgrade',
+        action='store_true',
+        default=False,
```
Comment (ksnavely) [inline src/cirrus/build.py]: I would argue we want this to be True, yeah?

File: src/cirrus/build.py [hunk 2]
```diff
@@ -100,16 +107,25 @@ def execute_build(opts):
...
+                '-r {2}'
+                ).format(venv_path, pypi_url, reqs_name)
 
     else:
         # no pypi server
-        cmd = '{0}/bin/pip install -r {1}'.format(venv_path, reqs_name)
+        if opts.upgrade:
+            cmd = '{0}/bin/pip install --upgrade -r {1}'.format(venv_path, reqs_name)
+        else:
+            cmd = '{0}/bin/pip install -r {1}'.format(venv_path, reqs_name)
```
Comment (ksnavely) [inline src/cirrus/build.py]: In both these cases can't you just append --upgrade to the command, so you DRY for the rest of it? Disregard if that's not the case.

**Comment by ksnavely** [comment]
LGTM

File: src/cirrus/build.py (hunk 1 above)
Comment (evansde77) [inline src/cirrus/build.py]: I dont want to magically change the default behaviour so preserving it for now

File: src/cirrus/build.py (hunk 2 above)
Comment (evansde77) [inline src/cirrus/build.py]: pip is picky about CLI arg ordering to some degree. Its ugly, but explicit

**Pull request merged**